### PR TITLE
GCI-203:Comparison of String object using != in Module.extensionsMatc…

### PR DESCRIPTION
**Pull request by Wandji69**
https://issues.openmrs.org/browse/GCI-203

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/Module.java [hunk 1]
```diff
@@ -530,11 +530,15 @@ public void setExtensionNames(Map<String, String> map) {
 
 	private boolean isNoNeedToExpand() {
 		if (extensionNames == null || extensionNames.isEmpty()) {
-			return true;
+			for(Extension ext : extensions) {
+				if (!extensionNames.get(ext.getPointId()).equals(ext.getClass().getName())) {
```
Comment (Andu033) [inline api/src/main/java/org/openmrs/module/Module.java]: Why try to get elements of extensionNames if the if already confirmed that is null or empty? In best case nothing happen and in worst case you get nullPointerException .

Comment (tenas97) [inline api/src/main/java/org/openmrs/module/Module.java]: I agree with Andu033. The if statement would be true if extensionNames is null. So trying to access extensionNames while inside the if statement(line 534) would probably lead to a nullPointerException, as is the case with the Travis test fail

Comment (Andu033) [reply api/src/main/java/org/openmrs/module/Module.java]: Thank you for strengthening my point of view.

**Comment by Wandji69** [comment]
Yes @Reyano i did run "mvn clean install"

**Comment by Reyano132** [comment]
@Wandji69 , code problem mention by @Andu033 is valid no doubt but  after when you run "mvn clean install" and you don't get any error then code definitely pass Travis CI build

**Comment by dkayiwa** [comment]
Can you respond to the review comments?

**Comment by Wandji69** [comment]
@Reyano132 Yes it pass travis CI build

**Comment by dkayiwa** [comment]
@Wandji69 can you add unit tests for your changes?

**Comment by Wandji69** [comment]
Okay I will look into it.

**Pull request closed without merging**